src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,972 @@
+# ==============================================================================
+# Psycopg Compatibility Shim
+#
+# REASON: In some CI/Linux environments, psycopg's pure-Python fallback is
+# installed instead of the C-optimized version. The pure-Python version has a
+# different default behavior: it returns `bytes` for certain metadata query
+# results (like schema names and data types), whereas the C version correctly
+# decodes them to `str`. This inconsistency breaks upstream libraries like Ibis.
+#
+# SOLUTION: This code block acts as a compatibility shim. It monkey-patches the
+# `psycopg.Cursor` class to normalize the data flow. It ensures that both
+# query parameters (on the way in) and query results (on the way out) are
+# consistently strings, regardless of which `psycopg` implementation is in use.
+# This patch is applied globally and exactly once when this module is imported.
```

Comment:
I don't feel good about patching something globally. It will definitely break things for users.